tests/python/gaia-ui-tests/gaiatest/apps/messages/app.py

```diff
@@ -52,20 +64,57 @@ def tap_first_received_message(self):
         from gaiatest.apps.messages.regions.message_thread import MessageThread
         return MessageThread(self.marionette)
 
+    def tap_thread_options(self):
+        self.marionette.find_element(*self._thread_options_button).tap()
+        from gaiatest.apps.messages.regions.options import ThreadOptions
+        return ThreadOptions(self.marionette)
+
+    def enter_select_mode(self):
+        edit_header = self.marionette.find_element(*self._edit_thread_header)
+        thread_options = self.tap_thread_options()
+        thread_options.select_threads()
+        Wait(self.marionette).until(lambda m: edit_header.rect['y'] == 0)
```

Comment:
I'll create a function to make the explanation clearer.

What happens here is: The header goes down. Once some pixels of the elements are displayed on the screen, `element.is_displayed()` returns `True`. So the wait will be over, even though the transition of the header is not done yet. Waiting on y == 0, allows us to know when the transition is done.